upgrade-2.x.sh

```diff
@@ -215,6 +215,34 @@ function _run_supervisor_update() {
     return "${ret}"
 }
 
+function _fetch_supervisor_version() {
+    local resp
+    local supervisor_version
+    local scheduled_supervisor_version
+
+    resp=$(CURL_CA_BUNDLE=${TMPCRT} curl --silent --retry 10 --header "Authorization: Bearer ${APIKEY}" "${API_ENDPOINT}/v6/device(uuid='${UUID}')?\$select=supervisor_version")
+    if supervisor_version=$(echo "${resp}" | jq -e -r '.d[0].supervisor_version'); then
+        if [ -z "${supervisor_version}" ]; then
+            log ERROR "Could not get current supervisor version from the API, got ${resp}"
+            return 1
+        fi
+        # Make sure there is no scheduled supervisor update
+        resp=$(CURL_CA_BUNDLE=${TMPCRT} curl --silent --retry 10 --header "Authorization: Bearer ${APIKEY}" "${API_ENDPOINT}/v6/device(uuid='${UUID}')?\$select=should_be_managed_by__supervisor_release&\$expand=should_be_managed_by__supervisor_release(\$select=supervisor_version)")
```

Comment:
@alexgg a) you could get just the supervisor_release's supervisor_version with:
```
curl "${API_ENDPOINT}/v6/supervisor_release?\$top=1&\$select=supervisor_version&\$filter=should_manage__device/any(d:d/uuid%20eq%20'${UUID}')"
```
but I think it would be better if you tried to combine both curls and fetch both the `device.supervisor_version` and the supervisor_release's supervisor_version in one request with
```suggestion
        resp=$(CURL_CA_BUNDLE=${TMPCRT} curl --silent --retry 10 --header "Authorization: Bearer ${APIKEY}" "${API_ENDPOINT}/v6/device(uuid='${UUID}')?\$select=should_be_managed_by__supervisor_release&\$expand=should_be_managed_by__supervisor_release(\$select=supervisor_version)")
        resp=$(CURL_CA_BUNDLE=${TMPCRT} curl --silent --retry 10 --header "Authorization: Bearer ${APIKEY}" "${API_ENDPOINT}/v6/device(uuid='${UUID}')?\$select=supervisor_version&\$expand=should_be_managed_by__supervisor_release(\$top=1;\$select=supervisor_version)")
```

That's both 1 less request & and is also faster then the query that starts from the `supervisor_release` (I expect that part of the reason is that the `supervisor_release` is a translation and also has some expensive permissions).

b) The API does support compressed request/responses, so given that it saves bandwidth I would use it.
The supervisor is doing so in all requests already for example and last year was even migrated to use brotli.